### PR TITLE
🐛 Source Intercom: Revert migration to low-code to prevent data loss

### PR DESCRIPTION
**Pull request by girarda**
## What
* Reverts the migration to low-code
* The contacts stream uses a stateful pagination mechanism, which isn't supported by low-code. This leads to dataloss if new records are creating while a sync is ongoing.
* This issue is tracked in https://github.com/airbytehq/airbyte/issues/26301
* This is not a breaking change since the spec and the schemas are unchanged
* This re-opens https://github.com/airbytehq/airbyte/issues/21335

## How
* Revert https://github.com/airbytehq/airbyte/commit/78828413a222f55fbd851a85accfb19124d95005
* Update README, Dockerfile, metadata.yaml, and setup.py

## Recommended reading order
1. `airbyte-integrations/connectors/source-intercom/Dockerfile`
2. `airbyte-integrations/connectors/source-intercom/metadata.yaml`
3. `airbyte-integrations/connectors/source-intercom/README.md`
4. Other files were a straight revert from https://github.com/airbytehq/airbyte/commit/78828413a222f55fbd851a85accfb19124d95005

## 🚨 User Impact 🚨
* Users won't lose records added when a sync is in progress

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-intercom/source_intercom/source.py [hunk 1]
```diff
@@ -2,17 +2,536 @@
 # Copyright (c) 2023 Airbyte, Inc., all rights reserved.
```
Comment (girarda) [inline airbyte-integrations/connectors/source-intercom/source_intercom/source.py:2]: this file was reset by reverting [this PR](https://github.com/airbytehq/airbyte/pull/23013)

File: airbyte-integrations/connectors/source-intercom/unit_tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#
```
Comment (girarda) [inline airbyte-integrations/connectors/source-intercom/unit_tests/conftest.py:1]: this file was reset by reverting [this PR](https://github.com/airbytehq/airbyte/pull/23013)

File: airbyte-integrations/connectors/source-intercom/unit_tests/test_cached_stream_state.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
+#
```
Comment (girarda) [inline airbyte-integrations/connectors/source-intercom/unit_tests/test_cached_stream_state.py:1]: this file was reset by reverting [this PR](https://github.com/airbytehq/airbyte/pull/23013)

File: airbyte-integrations/connectors/source-intercom/unit_tests/test_control_rate_limit.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+#
```
Comment (girarda) [inline airbyte-integrations/connectors/source-intercom/unit_tests/test_control_rate_limit.py:1]: this file was reset by reverting [this PR](https://github.com/airbytehq/airbyte/pull/23013)

File: airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py [hunk 5]
```diff
@@ -0,0 +1,339 @@
+#
```
Comment (girarda) [inline airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py:1]: this file was reset by reverting [this PR](https://github.com/airbytehq/airbyte/pull/23013)

File: airbyte-integrations/connectors/source-intercom/integration_tests/integration_test.py [hunk 6]
```diff
@@ -3,10 +3,139 @@
 #
```
Comment (girarda) [inline airbyte-integrations/connectors/source-intercom/integration_tests/integration_test.py:3]: this file was reset by reverting [this PR](https://github.com/airbytehq/airbyte/pull/23013)

File: airbyte-integrations/connectors/source-intercom/setup.py [hunk 7]
```diff
@@ -6,23 +6,23 @@
...
 
 MAIN_REQUIREMENTS = [
-    "airbyte-cdk==0.29.0",
+    "airbyte-cdk~=0.40",
 ]
 
 TEST_REQUIREMENTS = [
     "pytest",
-    "pytest-mock",
+    "requests-mock",
```
Comment (girarda) [inline airbyte-integrations/connectors/source-intercom/setup.py:14]: this is the dependency we used before [the migration to low-code](https://github.com/airbytehq/airbyte/commit/78828413a222f55fbd851a85accfb19124d95005#diff-b8515d841d0010d0b536b78ba5354491f42115d0982e7f95eb2cff299c8d10abR14)

File: airbyte-integrations/connectors/source-intercom/setup.py [hunk 8]
```diff
@@ -6,23 +6,23 @@
...
 setup(
     name="source_intercom",
-    description="Source implementation for Intercom Yaml.",
+    description="Source implementation for Intercom.",
     author="Airbyte",
     author_email="contact@airbyte.io",
     packages=find_packages(),
     install_requires=MAIN_REQUIREMENTS,
-    package_data={"": ["*.json", "*.yaml", "schemas/*.json", "schemas/shared/*.json"]},
+    package_data={"": ["*.json", "schemas/*.json", "schemas/shared/*.json"]},
```
Comment (girarda) [inline airbyte-integrations/connectors/source-intercom/setup.py:25]: we don't need yaml files anymore

**Comment by girarda** [comment]
moved back to draft as it's not clear this will solve the data loss issue

**Comment by girarda** [comment]
Update: @killthekitten [confirmed this wasn't an issue with the Python implementation so I think we should move forward with the revert.](https://github.com/airbytehq/airbyte/issues/26301#issuecomment-1590152756)

As additional context, the low-code implementation of source-intercom actually has a lot of custom code. Both [Python code](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-intercom/source_intercom/components.py), and [non-trivial jinja expressions](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-intercom/source_intercom/manifest.yaml#L269).

It's not obvious to me that we benefit much from the low-code framework for intercom.

**Comment by killthekitten** [comment]
@girarda I'd say it was actually a step up for us. For whatever reason our EU workspace didn't sync properly on the Python connector (I couldn't find the issue).

**Comment by killthekitten** [comment]
@girarda now I remembered another detail: we were never able to sync the EU workspace with the Python connector, only the US workspace worked.

So take my confirmation with a grain of salt.

**Comment by girarda** [comment]
closing because reverting isn't an obvious solution. we can always re-open if needed

**Pull request closed without merging**